Assets/Scripts/BehaviorTree/Node.cs

```diff
@@ -0,0 +1,88 @@
+using System.Collections.Generic;
+
+namespace BehaviorTree
+{
+    public enum NodeState
+    {
+        RUNNING,
+        SUCCESS,
+        FAILURE
+    }
+
+    public class Node
+    {
+        protected NodeState State;
+
+        public Node Parent;
```

Comment:
```Parent``` should not be settable. It should be a gettable property with a private ```set``` accessor.